### PR TITLE
fix(security)#Eval-438: fix security on BilanPeriodique's routes

### PR DESCRIPTION
**Pull request by Gregprod33**
## Describe your changes
Add security classes and ResourceFilters on BilanPeriodiqueController
## Checklist tests
Unit tests on security classes
Testing with and without rights setted on admin console
## Issue ticket number and link
[#EVAL-438](https://jira.support-ent.fr/browse/EVAL-438)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/fr/openent/competences/security/AccessConseilDeClasse.java [hunk 1]
```diff
@@ -14,7 +14,9 @@ public class AccessConseilDeClasse implements ResourcesProvider {
     @Override
     public void authorize(HttpServerRequest request, Binding binding, UserInfos user, Handler<Boolean> handler) {
         final String idStructure = request.params().get(Field.IDSTRUCTURE);
-        handler.handle(user.getStructures().contains(idStructure) &&
+        final String id_structure = request.params().get(Field.ID_STRUCTURE);
+        final String id_etablissement = request.params().get(Field.IDETABLISSEMENT);
```
Comment (Matiyeu) [inline src/main/java/fr/openent/competences/security/AccessConseilDeClasse.java:18]: Tu peux mettre tes noms de variables en camelcase

File: src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java [hunk 2]
```diff
@@ -281,7 +276,7 @@ public void handle(JsonObject synthese) {
      */
     @Post("/appreciation/CPE/bilan/periodique")
     @ApiDoc("Créer ou mettre à jour une appreciation CPE du bilan périodique d'un élève pour une période donnée")
-    @SecuredAction(value = "create.appreciation.CPE.bilan.periodique", type = ActionType.WORKFLOW)
+    @SecuredAction(value="create.appreciation.CPE.bilan.periodique", type=ActionType.WORKFLOW)
```
Comment (Matiyeu) [inline src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java:279]:
```suggestion
    @SecuredAction(value = "create.appreciation.CPE.bilan.periodique", type = ActionType.WORKFLOW)
```

File: src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java [hunk 3]
```diff
@@ -363,7 +359,8 @@ public void handle(UserInfos user) {
 
     @Post("/avis/bilan/periodique")
     @ApiDoc("Créer un avis de conseil de classe")
-    @SecuredAction(value = "create.avis.conseil.bilan.periodique", type = ActionType.AUTHENTICATED)
+    @SecuredAction(value="", type=ActionType.RESOURCE)
```
Comment (Matiyeu) [inline src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java:362]:
```suggestion
    @SecuredAction(value = "", type = ActionType.RESOURCE)
```

File: src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java [hunk 4]
```diff
@@ -394,7 +391,7 @@ public void handle(JsonObject opinion) {
 
     @Delete("/avis/bilan/periodique")
     @ApiDoc("Supprime un avis de conseil de classe")
-    @SecuredAction(value = "", type = ActionType.AUTHENTICATED)
+    @SecuredAction(value="create.avis.conseil.bilan.periodique", type=ActionType.WORKFLOW)
```
Comment (Matiyeu) [inline src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java:394]:
```suggestion
    @SecuredAction(value = "create.avis.conseil.bilan.periodique", type = ActionType.WORKFLOW)
```

File: src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java [hunk 5]
```diff
@@ -491,7 +489,7 @@ public void handle(JsonObject idAvisClasse) {
     @ResourceFilter(SetAvisConseilFilter.class)
     public void deleteAvisConseil(final HttpServerRequest request) {
         UserUtils.getUserInfos(eb, request, user -> {
-            if (user != null) {
+            if(user != null) {
```
Comment (Matiyeu) [inline src/main/java/fr/openent/competences/controllers/BilanPeriodiqueController.java:492]:
```suggestion
            if (user != null) {
```

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/fr/openent/competences/security/AccessConseilDeClasse.java (hunk 1 above)
Comment (Gregprod33) [reply src/main/java/fr/openent/competences/security/AccessConseilDeClasse.java:18]: Pour les magic String ?

Comment (Matiyeu) [reply src/main/java/fr/openent/competences/security/AccessConseilDeClasse.java:18]: Non tes variables final

**Comment by sjollois** [comment]
tu peux marquer les commentaires comme résolus une fois que tu as push le fix stp, pour une meilleur visibilité de la PR :)

**Pull request closed without merging**